src/model/registerConsent.model.ts

```diff
@@ -0,0 +1,328 @@
+    }
+    // channel name
+    return `RegisterConsent_${phase}_${id}`
+  }
+
+  static async triggerWorkflow (
+    phase: RegisterConsentPhase,
+    id: string,
+    pubSub: PubSub,
+    message: Message
+  ): Promise<void> {
+    const channel = RegisterConsentModel.notificationChannel(phase, id)
+    return deferredJob(pubSub, channel).trigger(message)
+  }
+
+  async onVerifyConsent (): Promise<void> {
+    // not sure what functions to use or if they are ready
+    // for now we are just going do nothing here.
+    // todo: update transition to
+    // - verify consent
+    // - store consent
+    // - throw errors if there are errors in verifying the consent
+    /*
+    const { consentsPostRequestAUTH, participantDFSPId } = this.data
+    try {
+    } catch (error) {
+      this.logger.push({ error }).error('start -> requestIsValid')
+
+      let mojaloopError
+      // if error is planned and is a MojaloopApiErrorCode we send back that code
+      if ((error as Errors.MojaloopApiErrorCode).code) {
+        mojaloopError = reformatError(error, this.logger)
+      } else {
+        // if error is not planned send back a generalized error
+        mojaloopError = reformatError(
+          Errors.MojaloopApiErrorCodes.TP_ACCOUNT_LINKING_ERROR,
+          this.logger
+        )
+      }
+
+      await this.thirdpartyRequests.putConsentsError(
+        consentsPostRequestAUTH.consentId,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        participantDFSPId
+      )
+
+      // throw error to stop state machine
+      throw error
+    }
+    */
+  }
+
+  async onRegisterAuthoritativeSourceWithALS (): Promise<void> {
+    const { consentsPostRequestAUTH, participantDFSPId } = this.data
+
+    // catch any unplanned errors and notify DFSP
+    try {
+      const waitOnParticipantResponseFromALSResponse = RegisterConsentModel.notificationChannel(
+        RegisterConsentPhase.waitOnParticipantResponseFromALS,
+        consentsPostRequestAUTH.consentId
+      )
+
+      await deferredJob(this.subscriber, waitOnParticipantResponseFromALSResponse)
+        .init(async (channel) => {
+          // todo: sdk-standard-components needs a postParticipantsTypeId function
+          //       building the request from scratch for now
+          const alsParticipantURI = `http://${this.config.alsEndpoint}/participants/CONSENT/${consentsPostRequestAUTH.consentId}`
+          const axiosConfig = {
+            headers: {
+              Accept: 'application/vnd.interoperability.participants+json;version=1.1',
+              'FSPIOP-Source': this.config.authServiceParticipantFSPId,
+              Date: (new Date()).toUTCString()
+            }
+          }
+          const payload: fspiopAPI.Schemas.ParticipantsTypeIDSubIDPostRequest = {
+            fspId: this.config.authServiceParticipantFSPId
+          }
+          const res = await axios.post(alsParticipantURI, payload, axiosConfig)
```

Comment:
Why not use this? https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/requests/baseRequests.js#L307

Reply:
We don't have type definitions for `BaseRequests` and instead of defining all the types this was easier until I can commit a  `postParticipantsTypeId` function to `sdk-standard-components`.